pandas/plotting/_matplotlib/converter.py

```diff
@@ -55,13 +56,42 @@ def get_pairs():
     return pairs
 
 
-def register(explicit=True):
-    # Renamed in pandas.plotting.__init__
-    global _WARN
+def pandas_converters_deco(func):
```

Comment:
I think `register_pandas_matplotlib_converters` could be a more clear name.

Also, if the context manager is not going to be used more than in the decorator, I would probably implement its code in the decorator (or move the context manager inside the decorator)/.